llvm/lib/Target/X86/X86InstrCMovSetCC.td

```diff
@@ -113,6 +113,35 @@ let Predicates = [HasCMOV, HasCF] in {
             (CFCMOV32rr GR32:$src1, (inv_cond_XFORM timm:$cond))>;
   def : Pat<(X86cmov GR64:$src1, 0, timm:$cond, EFLAGS),
             (CFCMOV64rr GR64:$src1, (inv_cond_XFORM timm:$cond))>;
+
+  def : Pat<(X86cload addr:$src1, 0, timm:$cond, EFLAGS),
+            (CFCMOV16rm addr:$src1, timm:$cond)>;
+  def : Pat<(X86cload addr:$src1, 0, timm:$cond, EFLAGS),
+            (CFCMOV32rm addr:$src1, timm:$cond)>;
+  def : Pat<(X86cload addr:$src1, 0, timm:$cond, EFLAGS),
+            (CFCMOV64rm addr:$src1, timm:$cond)>;
+
+  // FIXME: Shouldn't patterns for 0 work for undef?
```

Comment:
My understand regarding to https://github.com/llvm/llvm-project/pull/95515#issuecomment-2186613740
that we should not care about undef/posion in pattern match. They should be optimized out earlier.

Reply:
We can find some patterns for `insert_subvector, masked_load, masked_st` with undef operands in X86InstrAVX512.td llvm/lib/Target/X86/X86InstrSSE.td

Reply:
They seem all vector type undef. If we do need support both, I think there will be some pattern like zero_or_undef. But I didn't find it in current code base.

Reply:
TBH, I don't know. That's why I leave a FIXME.

Reply:
We can remove them for now.

Reply:
Done. Create zero node for undef instead.